ddtrace/internal/telemetry/metrics_namespaces.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any
+from typing import Dict
+
+from ddtrace.internal.constants import TELEMETRY_APPSEC
+from ddtrace.internal.constants import TELEMETRY_TRACER
+from ddtrace.internal.telemetry.metrics import CountMetric
+from ddtrace.internal.telemetry.metrics import GaugeMetric
+from ddtrace.internal.telemetry.metrics import Metric
+from ddtrace.internal.telemetry.metrics import MetricTagType
+from ddtrace.internal.telemetry.metrics import MetricType
+from ddtrace.internal.telemetry.metrics import RateMetric
+
+
+NamespaceMetricType = Dict[str, Dict[str, Any]]
+
+
+class TelemetryTypeError(Exception):
+    pass
+
+
+class MetricNamespace:
+    _metrics_data = {TELEMETRY_TRACER: {}, TELEMETRY_APPSEC: {}}  # type: NamespaceMetricType
+    metric_class = {"count": CountMetric, "gauge": GaugeMetric, "rate": RateMetric}
+
+    def _flush(self):
+        self._metrics_data = {TELEMETRY_TRACER: {}, TELEMETRY_APPSEC: {}}
+
+    def get(self):
+        # type: () -> Dict[str, Any]
+        namespace_metrics = self._metrics_data.copy()
+        return namespace_metrics
+
+    @staticmethod
+    def validate_type_metric(metric_check, metric_to_validate):
+        # type: (Metric, Metric) -> bool
+        if metric_check.__class__ == metric_to_validate.__class__:
+            return True
+        return False
+
+    def _add_metric(self, metric_type, namespace, name, value=1.0, tags={}, interval=10.0):
+        # type: (MetricType, str,str, float, MetricTagType, float) -> None
+        """
+        Queues metric
+        """
+        name = "dd.app_telemetry." + namespace + "." + name
+        metric_check = self.metric_class[metric_type](  # type: ignore[abstract]
+            namespace, name, tags=tags, common=True, interval=interval
+        )
+        metric = self._metrics_data[namespace].get(metric_check.id, metric_check)
+        if not self.validate_type_metric(metric_check, metric):
+            raise TelemetryTypeError(
```

Comment:
Is this error caught by the telemetry writer? Could this error crash an application?

Reply:
This error raises when you call `telemetry_writer.add_[type]_metric()`. In this case, a crash error it's ok to catch the problem while we're developing. If not, we've the risk to add a metric with errors/not working, merge and add it into a release